agents-common/src/main/java/org/apache/ranger/plugin/policyevaluator/RangerDefaultPolicyEvaluator.java

```diff
@@ -1512,4 +1504,27 @@ private List<RangerConditionEvaluator> createPolicyConditionEvaluators(RangerPol
 		return ret;
 	}
 
+	private static boolean isMatchForResourceMatchingScope(final RangerAccessRequest.ResourceMatchingScope scope, final RangerPolicyResourceMatcher.MatchType matchType, boolean isAnyMatch) {
+		boolean ret = false;
+		if (isAnyMatch){
+			ret = matchType !=  RangerPolicyResourceMatcher.MatchType.NONE;
+		}
+		else if (scope!=null) {
+				switch (scope) {
+						case SELF_OR_DESCENDANTS: {
+							ret = matchType != RangerPolicyResourceMatcher.MatchType.NONE;
+							break;
+						}
+						case SELF_AND_ALL_DESCENDANTS: {
+							ret = matchType != RangerPolicyResourceMatcher.MatchType.NONE;
+							break;
+						}
+						default: {
+							ret = matchType == RangerPolicyResourceMatcher.MatchType.SELF || matchType == RangerPolicyResourceMatcher.MatchType.SELF_AND_ALL_DESCENDANTS;
+							break;
+						}
+				}
+		}
```

Comment:
when scope is null, shouldn't the return be true if matchType is SELF or SELF_AND_ALL_DESCENDANTS?

Reply:
scope!=null always as per current implementation of RangerAccessRequestImpl .
`private ResourceMatchingScope resourceMatchingScope = ResourceMatchingScope.SELF;`
The default value is always SELF. But I see it can be somehow null if interface is implemented differently (I think we should prevent this)
Can there (or is there currently ) be a use case of null ResourceMatchingScope ?
However, I think a check for not null would be required for current refactoring logic as we cannot have null in switch case and we should return isMatched=False in this case unless there is an explicit use case for the same 
(if ResourceMatchingScope is null then I think isMatched should be false as matchType seems irrelevant).

Reply:
@fateh288  - the regression in handling scope==null should be addressed in this PR.

Reply:
Yes, I identified the issue is in test cases defined in test_policyengine_resource_with_req_expressions.json
The ResourceMatchingScope defined in these test cases is SELF_OR_CHILD which is not a valid value in the ResourceMatchingScope enum and results in null scope. 

I tried changing ResourceMatchingScope to SELF here and all the test cases pass here. Do you suggest doing this ? Or should we add SELF_OR_CHILD as a valid ResourceMatchingScope ?

Yes, I can handle null as a scope or prevent regressions, but ideally it should be an invalid scenario if implemented correctly.